clients/slack/slack_client.py

```diff
@@ -78,15 +78,16 @@ def upload_file(self, channel_name: str, file_path: str, message: SlackMessageSc
 
     def _get_channel_id(self, channel_name: str) -> str:
         try:
+            channels = self._get_channels()
             channel = [
```

Comment:
isn't it clearer to just iterate over the channels and return the first with a matching name?

Reply:
Well yes, but I wanted to make it more generic.
Searching by id means we will never return the full list of channels (which is fine), and I thought it will be nice to implement it in a way other could use to :)

Reply:
you can still get the channels from get_channels but here instead of iterating in a list comprehension you can just have a simple iteration